File: Java/1647. Minimum Deletions to Make Character Frequencies Unique.java
class Solution {
    public int minDeletions(String s) {
        int[] arr = new int[26];
        for(char c: s.toCharArray()){
            arr[c - 'a']++;
        }
        int res = 0;
        Arrays.sort(arr);
        for(int i = 24; i >= 0; i--){
            if(arr[i] == 0){
                break;
            }
            if(arr[i] >= arr[i+1]){
                int temp = arr[i];
                arr[i] = Math.max(0, arr[i+1] -1);
                res += temp - arr[i];
            }
        }
        return res;
    }
}

File: JavaScript/1647. Minimum Deletions to Make Character Frequencies Unique.js
/**
 * @param {string} s
 * @return {number}
 */
var minDeletions = function(s) {
    var arr = new Array(26).fill(0);
    for(var c of s){
        arr[c.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }
    var res = 0;
    arr.sort((a, b) => a-b);
    for(var i = 24; i >= 0; i--){
        if(arr[i] == 0){
            break;
        }
        if(arr[i] >= arr[i+1]){
            var temp = arr[i];
            arr[i] = Math.max(0, arr[i+1] -1);
            res += temp - arr[i];
        }
    }
    return res;
};
